homeassistant/components/tesla/binary_sensor.py

```diff
@@ -15,6 +15,21 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         for device in hass.data[TESLA_DOMAIN]["devices"]["binary_sensor"]
     ]
     add_entities(devices, True)
+    return True
+
+
+async def async_setup_entry(hass, config_entry, async_add_devices):
+    """Set up the Tesla binary_sensors by config_entry."""
+    return await hass.async_add_executor_job(
+        setup_platform, hass, config_entry.data, async_add_devices, None
```

Comment:
Pass an empty dict as discovery_info to setup_platform` and add a guard clause there that checks if discovery_info is None and return if so. We want to avoid that the platform setup is attempted via a platform config section.

Reply:
Ok